xgcm/grid.py

```diff
@@ -941,6 +941,27 @@ def _assign_metrics(self, metrics):
                 # check for duplicate dimensions among each axis metric
                 self._metrics[metric_axes].append(metric_var)
 
+    def _check_axes(self, axis):
+        """Checks if axis(or axes) in `axis` are part of the grid"""
+        check_axes = [ax for ax in axis if ax not in self.axes]
+        check_axes_str = ",".join(check_axes)
+        if any(check_axes):
+            raise ValueError("Axis %s not found in grid object" % check_axes_str)
```

Comment:
I'm not sure I understand the logic for splitting these things out. Isn't it an unnecessary layer of indirection. 

http://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection

Reply:
I think ill remove this. See below.